splinterd/src/config/partial.rs

```diff
@@ -14,11 +14,22 @@
 
 use std::time::Duration;
 
+/// ConfigSource displays the source of configuration values, used to identify which of the various
+/// config modules were used to create a particular PartialConfig object.
+#[derive(Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ConfigSource {
+    TomlFile,
+    Default,
+    EnvironmentVar,
```

Comment:
Consider calling this 'Environment'

Reply:
Eventually we may want to make this source per-field instead of per-config so we can fit the environment variable into the debug output. (Not in this PR though.)

Reply:
Agreed, the ConfigBuilder's build method could be simplified with having the source per-field.